airflow/listeners/listener.py

```diff
@@ -47,6 +50,15 @@ def __init__(self):
     def has_listeners(self) -> bool:
         return len(self.pm.get_plugins()) > 0
 
+    @property
```

Comment:
```suggestion
    @cached_property
```

Once we've computed this once per process it can't change again.